examples/blaze/src/main/scala/com/example/http4s/blaze/NestedResourceExample.scala

```diff
@@ -0,0 +1,31 @@
+package com.example.http4s.blaze
+
+import java.util.concurrent._
+import org.http4s._
+import org.http4s.client.Client
+import org.http4s.dsl._
+import org.http4s.server._
+import org.http4s.server.blaze._
+import org.http4s.client.blaze._
+import org.http4s.util.ProcessApp
+import scalaz._
+import scalaz.concurrent._
+import scalaz.stream._, Process._
+
+/** How to manage the lifecycle of a server and its dependent resources */
+object NestedResourceExample extends ProcessApp {
+  // This service depends on a client.
+  def service(client: Client): HttpService = HttpService {
+    case GET -> Root / "proxy" =>
+      client.toHttpService.run(Request(Method.GET, uri("http://http4s.org/")))
+  }
+
+  def main(args: List[String]) =
```

Comment:
Compare with #582.  Not as nice syntactically, but it works under more scenarios, and doesn't introduce such a weird type.

From the type of acquire, we sometimes know how to release.  Helpers may be possible here, but veer quickly into "out of scope" territory.  `ServerBuilder` provided an obvious home that doesn't exist for `Client` or `ExecutorService`.